target/linux/ath79/image/generic.mk

```diff
@@ -2229,6 +2237,31 @@ define Device/zbtlink_zbt-wd323
 endef
 TARGET_DEVICES += zbtlink_zbt-wd323
 
+define Device/zyxel_nwa112x
+  SOC := ar9342
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x050000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | \
+	lzma | uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGE_SIZE := 8192k
+  IMAGES += factory-$$$$(ZYXEL_MODEL_STRING).bin
+  IMAGE/factory-$$$$(ZYXEL_MODEL_STRING).bin := \
+	$$(IMAGE/sysupgrade.bin) | pad-to 8192k | zyxel-tar-bz2 \
+	vmlinux_mi124_f1e mi124_f1e-jffs2 | append-md5sum-bin
+endef
+
+define Device/zyxel_nwa1121-ni
+  $(Device/zyxel_nwa112x)
+  DEVICE_VENDOR := ZyXEL
```

Comment:
DEVICE_VENDOR might be moved to the common node (as it's Zyxel-specific as it appears)

Reply:
Agreed. The vendor is clearly visible from the included build recipe, thus vendor/model/variant are not scattered here.